### PR TITLE
Editorial: Simplify CreateArrayFromListOrRestricted

### PR DESCRIPTION
**Pull request by FrankYFTang**
address https://github.com/tc39/proposal-intl-locale-info/issues/45

@anba @sffc @zbraniecki @ryzokuken @gibson042

### CODE REVIEW

File: locale.html [hunk 1]
```diff
@@ -11,11 +11,9 @@ <h1>CreateArrayFromListOrRestricted ( _list_ , _restricted_ )</h1>
...
       </p>
       <emu-alg>
-        1. If _restricted_ is *undefined*, then
-          1. Return CreateArrayFromList( _list_ ).
-        1. Let _array_ be ! ArrayCreate(0).
-        1. Perform ! CreateDataPropertyOrThrow(_array_, "0", _restricted_).
-        1. Return _array_.
+        1. If _restricted_ is not *undefined*, then
+          1. Set _list_ to _restricted_ .
+        1. Return ! CreateArrayFromList(  &laquo; _restricted_ &raquo; ).
```
Comment (anba) [inline locale.html:16]:
```suggestion
        1. Return ! CreateArrayFromList( _list_ ).
```

File: locale.html [hunk 2]
```diff
@@ -11,11 +11,9 @@ <h1>CreateArrayFromListOrRestricted ( _list_ , _restricted_ )</h1>
         The abstract operation CreateArrayFromListOrRestricted accepts the arguments _list_ and _restricted_ , and performs the following steps:
       </p>
       <emu-alg>
-        1. If _restricted_ is *undefined*, then
-          1. Return CreateArrayFromList( _list_ ).
-        1. Let _array_ be ! ArrayCreate(0).
-        1. Perform ! CreateDataPropertyOrThrow(_array_, "0", _restricted_).
-        1. Return _array_.
+        1. If _restricted_ is not *undefined*, then
+          1. Set _list_ to _restricted_ .
```
Comment (anba) [inline locale.html:15]:
```suggestion
          1. Set _list_ to &laquo; _restricted_ &raquo;.
```

**Review by anba — CHANGES_REQUESTED** [review]

**Comment by FrankYFTang** [comment]
PTAL

**Review by anba — APPROVED** [review]
LGTM!

**Pull request merged**